webapp/src/services/base.service.js

```diff
@@ -1,10 +1,31 @@
+import axios from "axios";
+import history from "../History";
+
 export function authHeader() {
-    // return authorization header with basic auth credentials
-    let user = JSON.parse(localStorage.getItem("user"));
-  
-    if (user) {
-      return { "Authorization": user.token };
-    } else {
-      return {};
-    }
-  }
\ No newline at end of file
+  // return authorization header with basic auth credentials
+  let user = JSON.parse(localStorage.getItem("user"));
+
+  if (user) {
+    return { Authorization: user.token };
+  } else {
+    return {};
+  }
+}
+
+axios.interceptors.response.use(
+  response => {
+    return response;
+  },
+  function(error) {
+    if (
+      error &&
+      error.response &&
+      error.response.status === 401 &&
+      error.response.data &&
+      error.response.data.type === "UNAUTHORIZED"
+    ) {
+      localStorage.removeItem("user");
```

Comment:
We also need to signal (somehow?) to the UserDropdown component that the user has been logged out...

Reply:
Currently working on this, it is a little more complicated than I expected.

Reply:
So I sorted this issue. 

I tried adding a listener to localstorage changes and then this updates user in the User.js, but the listener wasn't been triggered from changes to localstorage from axios. 

I found a work around, but just always checking localstorage and comparing it to the props received.

Quite a trick problem, redux was made to solve these kinds of issues, with communicating state changes to a wide variety of components, but it works so it fine for now :)